src/components/Attachments/AttachmentsSagas.js

```diff
@@ -0,0 +1,139 @@
+import { eventChannel, END } from 'redux-saga';
+import { all, put, call, fork, take, takeEvery } from 'redux-saga/effects';
+import { ResumableUploadToGoogleDrive } from 'libs/google-drive-uploader';
+import auth from 'auth';
+import * as Actions from './AttachmentsActions';
+
+const createUploaderCallback = emitter => (result, error) => {
+  if (error) {
+    emitter({
+      type: 'error',
+      error,
+    });
+    emitter(END);
+    return;
+  }
+
+  const handlersMap = {
+    initialize: () => emitter({ type: 'initialized' }),
+    getLocation: () => emitter({ type: 'getLocation' }),
+    start: () => emitter({ type: 'started' }),
+    Uploading: () =>
+      emitter({
+        type: 'uploading',
+        progress: result.progressNumber.current / result.progressNumber.end,
+      }),
+    Done: () => {
+      emitter({
+        type: 'done',
+        result: result.result,
+      });
+      emitter(END);
+    },
+  };
+  const handleResult = handlersMap[result.status];
+
+  if (!handleResult) {
+    // @note: we don't need really handle this exception
```

Comment:
So why are we handling it? I'm confused :)

Reply:
we need to make an exception for developer - if he makes mistake when modify the code

but in fact, there shouldn't be exceptions here and I mean that we shouldn't handle this exception in outer scope